### PR TITLE
[SDKS-7541] Add getTreatments for sets

### PR DESCRIPTION
**Pull request by nmayorsplit**
- Create method GetTreatmentsByFlagSet
- Create method GetTreatmentsByFlagSets
- Create method GetTreatmentsWithConfigByFlagSet
- Create method GetTreatmentsWithConfigByFlagSets

### CODE REVIEW

File: client/src/main/java/io/split/client/SplitClientImpl.java [hunk 1]
```diff
@@ -341,6 +371,86 @@ private Map<String, SplitResult> getTreatmentsWithConfigInternal(String matching
...
+            return new HashMap<>();
+        }
+        FlagSetsValidResult flagSetsValidResult = FlagSetsValidator.areValid(sets);
+        try {
+            if (!flagSetsValidResult.getValid()) {
+                _log.warn("The sets are invalid");
+                return new HashMap<>();
+            }
+            if (filterSetsAreInConfig(flagSetsValidResult.getFlagSets()).isEmpty()) {
+                _log.warn("The sets are not");
```
Comment (gthea) [inline client/src/main/java/io/split/client/SplitClientImpl.java:388]: Something seems to be missing from the message

File: client/src/main/java/io/split/client/SplitClientImpl.java [hunk 2]
```diff
@@ -341,6 +371,86 @@ private Map<String, SplitResult> getTreatmentsWithConfigInternal(String matching
...
+                return new HashMap<>();
+            }
+            checkSDKReady(methodEnum);
+            if (_container.isDestroyed()) {
+                _log.error("Client has already been destroyed - no calls possible");
+                return createMapControl(getAllFlags(flagSetsValidResult.getFlagSets()));
+            }
```
Comment (gthea) [inline client/src/main/java/io/split/client/SplitClientImpl.java:395]: I think the matchingKeyValidation is missing
```suggestion
            }
            
            if (!KeyValidator.isValid(matchingKey, "matchingKey", _config.maxStringLength(), methodEnum.getMethod())) {
            return createMapControl(getAllFlags(flagSetsValidResult.getFlagSets()));
        }
```

File: client/src/main/java/io/split/client/SplitClientImpl.java
Comment (gthea) [inline client/src/main/java/io/split/client/SplitClientImpl.java:1]: Maybe I would suggest extracting a method for the functionality from [here](https://github.com/splitio/java-client/pull/443/files#diff-ebfd38b8cdc8335ac335dd95993624d2bc92516f44ba8791c1033444231d33dbR323) to [here](https://github.com/splitio/java-client/pull/443/files#diff-ebfd38b8cdc8335ac335dd95993624d2bc92516f44ba8791c1033444231d33dbR355), and reusing it in both `getTreatmentsWithSetsAndWithConfigInternal` and `getTreatmentsWithConfigInternal`.

File: client/src/test/java/io/split/client/SplitClientImplTest.java [hunk 3]
```diff
@@ -47,10 +47,10 @@
...
 import java.util.concurrent.TimeoutException;
 
-import static org.hamcrest.Matchers.equalTo;
-import static org.hamcrest.Matchers.is;
-import static org.hamcrest.Matchers.nullValue;
-import static org.junit.Assert.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
```
Comment (gthea) [inline client/src/test/java/io/split/client/SplitClientImplTest.java:53]: 👍 👍

File: client/src/main/java/io/split/client/SplitClientImpl.java (hunk 1 above)
Comment (nmayorsplit) [reply client/src/main/java/io/split/client/SplitClientImpl.java]: Updated.

File: client/src/main/java/io/split/client/SplitClientImpl.java (hunk 2 above)
Comment (nmayorsplit) [reply client/src/main/java/io/split/client/SplitClientImpl.java]: Updated.

File: client/src/main/java/io/split/client/SplitClientImpl.java
Comment (nmayorsplit) [reply client/src/main/java/io/split/client/SplitClientImpl.java:1]: Updated.

**Review by gthea — APPROVED** [review]

**Pull request merged**